### DISCUSSION TITLE
Get Access Token Shortcut Requires Phone To Be Unlocked?

### DISCUSSION THREAD

**Discussion by jumpgap** [TeslaBuds/AuthAppForTesla#14] [General]
Hello,

It appears in the most recent update, the Shortcut to Get Access Token will not run in the background when you phone is locked. Was this an intended change? Looking or a way around it.

Thanks!

**Comment by Kemmey** [comment]
Yeah, that was an intended change. It was a major oversight, triggered by Xcode itself resetting these settings, that it ever shipped without. Having access to the token without Keychain being unlocked should not work at all. 

Can you describe scenarios where this hinders you? On the watch for Watch app for Tesla I only require unlock after first boot. Could be a less secure option here too.

**Reply by jumpgap** [reply]
Thanks for responding. I use your app to create a series of Shortcuts to do various things. On the lowest level, basic functions like open the truck. It is convenient to pull my phone out at the grocery store and trigger this with Siri. If I'm wearing a mask, it is often locked. I now need to unlock the phone and then trigger the Siri shortcut. I would say this is a minor issue.

On a bigger issue, I use the Shortcuts automation to trigger things at different times of the day. In the morning, I look to see if it is a weekday and it is cooler than 60 degrees F outside. If it is, it turns on the climate controls. I have another automation that runs every day at 11:00. Monday through Friday, it sets the charge level of my car to 60%. For the weekends, it sets it to 90% for me. None of these shortcuts will run now because of the need to unlock the phone.

I understand your concern for security. However, there is a significant loss of functionality within Shortcuts when you require it to run with the phone unlocked.

As a compromise, would you be willing to add a switch to turn this feature on or off. This way it would be a decision made by the end user to decide how they want it to work. I would also support going back to the old way too.

Thanks for taking the time to dig into this.

Regards,
Jim

**Reply by Kemmey** [reply]
I get it - I'll actually just remove the restriction in next release - you can grab the TestFlight beta now here: 

https://testflight.apple.com/join/jTixIetK

**Comment by jumpgap** [comment]
Thank you! Got the Test Flight release. Works great!

Regards,
Jim
